### DISCUSSION TITLE
Null output parameters

### DISCUSSION THREAD

**Discussion by wolski0420** [rabbitmq/rabbitmq-perf-test#354] [Q&A]
I'm load testing multiple rabbit servers. I'm using -o flag to dump results to the local file. I've noticed that I have one data column filled by NULL and it is "returned (msg/s)" column. 

![image](https://user-images.githubusercontent.com/55734622/178147125-4ade0f61-b362-415a-a49d-a57c805d14a2.png)

I have some questions:
1. What does it exactly mean "returned (msg/s)". Is it "number of messages returned to the rabbit server from clients (sender/receiver) per second"?
2. How can I enable it? Should I use some additional flags? Or it just depends on test situation?
3. Is it connected with NACKing messages? If yes - how? (my reasoning is, if client nack received message, it is going back to the server what implies "returned message"). I'm asking because I've noticed that "nack (msg/s)" column is not NULLed but has 0 in all rows.

I'm not advanced in RabbitMQ protocol. At this moment I'm trying to understand the behavior of Rabbit basing on load tests provided by PerfTest.

**Comment by michaelklishin** [comment]
Your questions are covered in [Unroutable Message Handling](https://rabbitmq.com/publishers.html#unroutable).
